### PR TITLE
Disable Cloudformation Stack  termination protection 

### PR DESCRIPTION
**Pull request by etsangsplk**
- Cloudformation Stack with termination protection will prevent stack from being deleted.
- Add action to disable termination protection. user can choose to use this before delete stack or just as is for other reason
- Added tests

### CODE REVIEW

**Comment by kapilt** [comment]
Thanks for the pull request, would you mind signing our contributor license agreement, referenced from the bottom of the project readme, direct link https://docs.google.com/forms/d/19LpBBjykHPox18vrZvBbZUcK6gQTj7qv1O5hCduAZFU/viewform

**Comment by etsangsplk** [comment]
@kapilt Done

File: c7n/resources/cfn.py [hunk 1]
```diff
@@ -72,6 +72,41 @@ def process_stacks(self, stack):
...
+                  - StackStatus: CREATE_COMPLETE
+                actions:
+                  - disable-protection
+    """
+
+    schema = type_schema('disable-protection')
+    permissions = ('cloudformation:UpdateStack',)
+
+    def process(self, stacks):
+        with self.executor_factory(max_workers=10) as w:
```
Comment (kapilt) [inline c7n/resources/cfn.py]: the concurrency here seems a bit high, i'd suggest dialing this back to like 3.

File: c7n/resources/cfn.py [hunk 2]
```diff
@@ -72,6 +72,41 @@ def process_stacks(self, stack):
         client.delete_stack(StackName=stack['StackName'])
 
 
+@CloudFormation.action_registry.register('disable-protection')
+class DisableProtection(BaseAction):
+    """Action to disable termination protection
```
Comment (kapilt) [inline c7n/resources/cfn.py]: fwiw, generally we've trying to move away from disable/enable style towards a `set-protection` that can cover off on both use cases, it helps reduce the size of the vocabulary that users have to learn.

File: tests/test_cfn.py [hunk 3]
```diff
@@ -39,6 +39,38 @@ def test_query(self):
         resources = p.run()
         self.assertEqual(resources, [])
 
+    def test_disable_protection(self):
+        factory = self.replay_flight_data('test_cfn_disable_protection')
+        p = self.load_policy({
```
Comment (kapilt) [inline tests/test_cfn.py]: none of these tests seems like their actually testing anything in the pr, ie. the action implementation could be replaced with a no-op and they would still pass.

**Comment by kapilt** [comment]
i'm starting to see at least one of the testing issues here is due to the api, the aws api is a bit broken wrt to its own documentation, per the docs, stack protection status should be reported by describe stacks, but i've created a protected stack, and it doesn't report the status **unless** one describes a single stack in isolation.

ie.
```
$ aws cloudformation describe-stacks
{
    "Stacks": [
        {
            "StackId": "arn:aws:cloudformation:us-east-2:11112222:stack/mytopic/7efb2d40-3a5d-11e8-8c44-500cef930ce6",
            "StackName": "mytopic",
            "CreationTime": "2018-04-07T12:16:27.160Z",
            "RollbackConfiguration": {},
            "StackStatus": "CREATE_COMPLETE",
            "DisableRollback": false,
            "NotificationARNs": [],
            "Tags": []
        }
    ]
}
$ aws cloudformation describe-stacks --stack-name mytopic
{
    "Stacks": [
        {
            "StackId": "arn:aws:cloudformation:us-east-2:11112222:stack/mytopic/7efb2d40-3a5d-11e8-8c44-500cef930ce6",
            "StackName": "mytopic",
            "CreationTime": "2018-04-07T12:16:27.160Z",
            "RollbackConfiguration": {},
            "StackStatus": "CREATE_COMPLETE",
            "DisableRollback": false,
            "NotificationARNs": [],
            "Tags": [],
            "EnableTerminationProtection": true
        }
    ]
}
```

**Comment by kapilt** [comment]
i've pushed updates per review comments.

**Comment by etsangsplk** [comment]
https://github.com/aws/aws-cli/issues/3247

**Pull request merged**